common/middleware/getOffenderDetails.test.js

```diff
@@ -0,0 +1,82 @@
+const getOffenderDetails = require('./getOffenderDetails')
+const { getOffenderData } = require('../data/sentencePlanningApi')
+
+jest.mock('../../common/data/sentencePlanningApi')
+
+describe('getOffenderDetails middleware', () => {
+  let mockOffenderData
+  let req
+  let res
+  const next = jest.fn()
+  const render = jest.fn()
+  const id = '0118 999 881 999 119 7253'
+  const tokens = {}
+  beforeEach(() => {
+    mockOffenderData = {
+      familyName: 'Shakey',
+      forename1: 'Bernard',
+      crn: 'S000001',
+      nomisId: 'A0000AB',
+      unnecessary: 'values',
+    }
+    req = { tokens, params: { id } }
+    res = { render, locals: {} }
+    getOffenderData.mockResolvedValue(mockOffenderData)
+  })
+  afterEach(() => {
+    getOffenderData.mockReset()
+    next.mockReset()
+    render.mockReset()
+  })
+  describe('with complete data', () => {
+    beforeEach(async () => {
+      await getOffenderDetails(req, res, next)
+    })
+    it('should return an object with the required values', () => {
+      expect(res.locals.offenderDetails).toEqual({
+        fullName: 'Bernard Shakey',
+        crn: 'S000001',
+        noms: 'A0000AB',
+      })
+    })
+    it('should call the data service once and pass the id', () => {
+      expect(getOffenderData).toHaveBeenCalled()
```

Comment:
Update expectation to show it has indeed been called once? 

`expect(getOffenderData).toHaveBeenCalledTimes(1)`

Reply:
I think I could replace all the `toHaveBeenCalled` methods with `toHaveBeenCalledTimes(1)` 🤔 I'll do that 👍